server/src/main/java/io/crate/execution/dml/ObjectIndexer.java

```diff
@@ -96,11 +96,13 @@ public void indexValue(@NotNull Map<String, Object> value, IndexDocumentBuilder
             Child child = entry.getValue();
             Object innerValue = null;
             if (value.containsKey(innerName) == false) {
+                // Generated value is not provided, nothing to check, use generated value.
                 innerValue = docBuilder.getSyntheticValue(child.ident());
```

Comment:
Actually, I meant to target only generated value verification. 
This is column constraint, so we are not omitting CHECK constraint. 

Looks like NOT NULL constraint got skipped, will double check.

Reply:
```
execute("create table t (id int, obj object as (a object as (b int not null)))");
execute("insert into t (id, obj) (select 1, {\"a\" = {}} from sys.cluster)");
assertThat(response.rowCount()).isEqualTo(0L);
```

works both on master and on this PR. This is because NOT NULL is triggered as a table constraint because it's not in target columns. Looks like in general it's not possible to have sub-column column constraint NOT NULL but I will push a change to target only generated column, we anyway don't want to skip NOT NULL check

Reply:
Pushed a change to be more future-proof. 

I can't think of example where first commit state would be failing but let's keep it safe and skip only generated not provided.